run_querytns.py

```diff
@@ -0,0 +1,225 @@
+&include_frb=0&name=&name_like=0&isTNS_AT=all&public=all&ra=&decl=&radius=\
+&coords_unit=arcsec&reporting_groupid[]=null&groupid[]=null\
+&classifier_groupid[]=null\
+&objtype[]={}\
+&at_type[]=null\
+&date_start[date]={}\
+&date_end[date]={}&discovery_mag_min=&discovery_mag_max=\
+&internal_name=&discoverer=&classifier=&spectra_count=&redshift_min={}\
+&redshift_max={}&hostname=&ext_catid=&ra_range_min=&ra_range_max=\
+&decl_range_min=&decl_range_max=&discovery_instrument[]=null\
+&classification_instrument[]=null&associated_groups[]=null\
+&at_rep_remarks=&class_rep_remarks=&frb_repeat=all&frb_repeater_of_objid=\
+&frb_measured_redshift=0&frb_dm_range_min=&frb_dm_range_max=&frb_rm_range_min=\
+&frb_rm_range_max=&frb_snr_range_min=&frb_snr_range_max=&frb_flux_range_min=\
+&frb_flux_range_max=&num_page=500&display[redshift]=1&display[hostname]=1\
+&display[host_redshift]=1&display[source_group_name]=1\
+&display[classifying_source_group_name]=1&display[discovering_instrument_name]=0\
+&display[classifing_instrument_name]=0&display[programs_name]=0\
+&display[internal_name]=1&display[isTNS_AT]=0&display[public]=1\
+&display[end_pop_period]=0&display[spectra_count]=1\
+&display[discoverymag]=1&display[discmagfilter]=1\
+&display[discoverydate]=1&display[discoverer]=1\
+&display[remarks]=0&display[sources]=0&display[bibcode]=0\
+&display[ext_catalogs]=0".format(months,objtype,date_begin,date_end,z_min,z_max)
+
+logger.info('Date begin = {}, date_end = {}'.format(date_begin,date_end))
+logger.debug(query)
+
+# Query TNS
+logger.info('querying TNS for all targets, this may take awhile')
+con = requests.post(query)
+con.raise_for_status()
+logger.info('query successful')
+# Parse output using BS4, find html table
+soup = BeautifulSoup(con.text, 'html.parser')
+tab = soup.find("table",'results-table')
+logger.info('HTML table found')
+
+# Get SN names from TNS html table
+def get_names(soup):
+    names = soup.find_all('td','cell-name')
+    names_list = []
+    for name in names:
+        if name.text.startswith('SN'):
+            names_list.append(name.text.replace(' ',''))
+    return names_list
+nms = get_names(tab)
+
+# Remove already existing names using flows api
+included_names = []
+flows_targets = api.get_targets()
+for target in flows_targets:
+    included_names.append('SN'+target['target_name'])
+nms = list(set(nms)-set(included_names))
+logger.info('Target names obtained: {}'.format(nms))
+
+# Query TNS for object info using API, then upload to FLOWS using API.
+sleep_time = 0.25 # seconds to wait in between each query
+for name in nms:
+    sn = name.replace('SN','')
+    logger.info('querying TNS for: {}'.format(sn))
+
+    # make GET request to TNS via API
+    get_obj = [("objname",sn), ("photometry","0"), ("spectra","0")]
+    response = get(url_tns_api,get_obj)
+
+    # Parse output
+    if None not in response:
+        if response.status_code not in [200,201,429]:
```

Comment:
Just do `response.raise_for_status()` which will raise a HTTPError if the response was unsuccessful. This is kind of doing manual checking on top of that...

Reply:
done